pkg/webhook/mutation/pod/metadata/workload.go

```diff
@@ -102,6 +106,11 @@ func findRootOwner(ctx context.Context, clt client.Client, partialObjectMetadata
 		}
 	}
 
+	if !foundController {
```

Comment:
In such a case :
```
pod < controller A < controller B < owner
```
the current way - function returns B
fixed way - function returns UNKNOWN